pkg/apis/ocs/v1/storagecluster_types.go

```diff
@@ -70,8 +70,14 @@ type ManageCephFilesystems struct {
 
 // ManageCephObjectStores defines how to reconcile CephObjectStores
 type ManageCephObjectStores struct {
-	ReconcileStrategy   string `json:"reconcileStrategy,omitempty"`
-	DisableStorageClass bool   `json:"disableStorageClass,omitempty"`
+	ReconcileStrategy   string        `json:"reconcileStrategy,omitempty"`
+	DisableStorageClass bool          `json:"disableStorageClass,omitempty"`
+	ManageGateway       ManageGateway `json:"manageGateway,omitempty"`
```

Comment:
Interesting thought to add the substructure for `ManageGateway` - do you anticipate other settings for Gateway? Otherwise directly having `GatewayInstances`inside the `ManageObjectStore` struct would also have been sufficient. @jarrpa - what do you think?

Reply:
Hi @obnoxxx, I agree that we could simply put all ObjectStore config settings directly into the ManageObjectStore struct. It was my thought to follow the form of the rook cephv1 struct, to build familiarity and ease of use for users of both projects. Still, this is not necessary at all. I'm happy to change this if we think it's better to just flatten these management structs. It would certainly simplify the interface, and whatever we do here, we should likely follow form later, so we should certainly get it right.

Reply:
I'm in favor of having it directly in the ManageCephObjectStores struct. We want to avoid mimicking the underlying APIs where possible, in favor of a cleaner API informed by our support statements. My understanding is that there is only ever one CephObjectStore per CephCluster, as well.

Reply:
I am also inclining to omitting the extra layer.

I don't fully understand how the 1:1 correspondence would have an impact on this though:
Instead of having a sub-structure for the gateway options, we would need to have a list of gateway options (or a list of substructures). Or do you mean that a list of substructures would be more natural and easy for multiple object stores?

Reply:
I'm good with just keeping this in a flat struct. I was close to my indifference point on this anyway.

Reply:
this has been taken care of